### ISSUE TITLE
Optional path parameters plus validation always fail when the parameter is ommited.

### ISSUE THREAD

**Issue by stevenj** [poem-web/poem#650]
## Expected Behavior

Optional path parameters shouldn't be validated if they are not present.

## Actual Behavior

Validation fails, because the parameter is missing.

## Steps to Reproduce the Problem

I defined a path parameter like so:

```rust
       #[oai(validator(
            pattern = "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}",
            max_length = "36"
        ))]
        action: Path<Optional<String>>,
```

It will only validate if present, so it's not optional.

**Comment by sunli829** [comment]
The path parameter cannot be optional, but I did a test and couldn't reproduce the problem you encountered.

```rust
     struct Api;

    #[OpenApi]
    impl Api {
        #[oai(path = "/", method = "get")]
        async fn test(
            &self,
            #[oai(validator(
                pattern = "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}",
                max_length = "36"
            ))]
            action: Path<Option<String>>,
        ) {
            assert_eq!(action.0, None);
        }
    }

    let api = OpenApiService::new(Api, "test", "1.0");
    TestClient::new(api)
        .get("/")
        .send()
        .await
        .assert_status_is_ok();
```

**Comment by stevenj** [comment]
You are correct about path parameters not being allowed to be optional in OpenAPI.  
The difference in our examples is I declared the path as `/my/path/:optional_thing`.  Which isn't clear from my bug report.  Apologies.

However given that they are not optional, then I have no issue with the behavior I experienced.
The only comment I could make if it's possible, would be to restrict the type of `T` in `Path<T>` to something with isn't optional, so that code like I attempted wouldn't even compile.  I haven't tested it but perhaps:

```rust
/// Represents the parameters passed by the URI path.
pub struct Path<T>(pub T) {
    _marker: PhantomData<T>,
};

impl<T> Path<T> where T: std::panic::UnwindSafe {
    // Empty implementation block
    // Path<T> can not be Path<Option<T>
}
```

might work?

**Comment by stevenj** [comment]
You are correct about path parameters not being allowed to be optional in OpenAPI.
The difference in our examples is I declared the path as /my/path/:optional_thing. Which isn't clear from my bug report. Apologies.

However given that they are not optional, then I have no issue with the behavior I experienced.
The only comment I could make if it's possible, would be to restrict the type of T in Path<T> to something with isn't optional, so that code like I attempted wouldn't even compile, or check it at runtime or when the schema is being built?
